Code/Framework/AzCore/AzCore/IO/Streamer/RequestPath.cpp

```diff
@@ -121,19 +108,19 @@ namespace AZ
                 AZ_Assert(FileIOBase::GetInstance(),
                     "Trying to resolve a path in RequestPath before the low level file system has been initialized.");
 
-                char fullPath[AZ::IO::MaxPathLength];
-                if (!FileIOBase::GetInstance()->ResolvePath(m_path.c_str(), fullPath, AZ_ARRAY_SIZE(fullPath)))
+                AZStd::optional<AZ::IO::FixedMaxPath> fullPath = FileIOBase::GetInstance()->ResolvePath(m_path);
```

Comment:
FYI we do have a[ ResolvePath overload](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/FileIO.h#L250-L251) that accepts a FixedMaxPath by reference as well

Reply:
I'm a bit concerned about in-place replacements causing problems as there are several implementations of that call, but ended up going for it anyway.